### DISCUSSION TITLE
Refactoring the Setting page.

### DISCUSSION THREAD

**Discussion by ildyria** [LycheeOrg/Lychee#2584] [Ideas]
### Enhancement

Currently the setting page does not allow the Dev to order the categories and settings in a more structured way.
Proposition to:
1. Create a `Categories` table with `category`, `order`.
2. Link `Configs` table to `Categories` via the `cat` column.
3. Reorder some settings, rename some categories.
4. Add missing descriptions / details.
5. Make some settings hidden in docker installations (set by a `.env` variable) because they are not relevant.
6. Verify that the Legacy response still works as expected.

**Comment by d7415** [comment]
A filter/search box that filters the list as you type may also be useful
